source/s3_auto_ranged_get.c

```diff
@@ -196,6 +196,15 @@ static bool s_s3_auto_ranged_get_update(
                     request->part_range_end = meta_request->part_size - 1; /* range-end is inclusive */
                     request->discovers_object_size = true;
 
+                    struct aws_s3_buffer_pool_ticket *ticket 
+                        = aws_s3_buffer_pool_reserve(meta_request->client->buffer_pool, meta_request->part_size);
+
+                    if (ticket == NULL) {
+                        goto has_work_remaining;
```

Comment:
If we have simultaneous meta-requests with different-sized parts, is there an issue where the pool will start rejecting large parts, but still let small parts through, so the large-parts get stuck forever with small-parts cutting in front in line

we might need some system to prevent this?

like, once ANY reservation fails in an update() loop, then all subsequent reservations should fail?

We could fast-follow with a fix for this, since it's pretty difficult at the moment to have different meta-requests using different sized parts (but I know python would like it easily customizable at a per-meta-request level)

Reply:
Discussed offline. Added concept of reservation hold thats placed on buffer whenever reservation fails and cleared on the following round of scheduling.